modules/cjson/Kconfig

```diff
@@ -9,3 +9,6 @@ config CJSON_LIB
 	# Since minimal libc lacks sscanf and strtod we need to use newlibc
 	# Allow external libc to allow native posix to be used for unit test
 	depends on NEWLIB_LIBC || EXTERNAL_LIBC
+	# cJSON library is using snprintf with floats. We therefore need to
+	# enable NEWLIB_LIBC_FLOAT_PRINTF to ensure that these are handled correctly.
+	imply NEWLIB_LIBC_FLOAT_PRINTF if NEWLIB_LIBC
```

Comment:
> \# cJSON library is using snprintf with floats. We therefore need to
> \# enable NEWLIB_LIBC_FLOAT_PRINTF to ensure that these are handled correctly.

This sounds like `imply` is the wrong choice here, as `imply` allows users to disable `NEWLIB_LIBC_FLOAT_PRINTF`.

You already have a `depends on NEWLIB_LIBC` above, so should this be `depends on NEWLIB_LIBC_FLOAT_PRINTF` to be aligned.

Reply:
Sounds reasonable, I will update the PR to `depends on`.

Reply:
Probably need to enable this explicitly in the project configuration then

Reply:
Yes, the plan was to add it to the `lte_ble_gateway` PR.

Reply:
Both are fine